src/libraries/System.Private.CoreLib/src/System/Byte.cs

```diff
@@ -332,6 +333,72 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
         /// <inheritdoc cref="IBinaryInteger{TSelf}.TrailingZeroCount(TSelf)" />
         public static byte TrailingZeroCount(byte value) => (byte)(BitOperations.TrailingZeroCount(value << 24) - 24);
 
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.TryReadBigEndian(ReadOnlySpan{byte}, bool, out TSelf)" />
+        static bool IBinaryInteger<byte>.TryReadBigEndian(ReadOnlySpan<byte> source, bool isUnsigned, out byte value)
+        {
+            byte result = default;
+
+            if (source.Length != 0)
+            {
+                if (!isUnsigned && sbyte.IsNegative((sbyte)source[0]))
+                {
+                    // When we are signed and the sign bit is set is set we are negative and therefore
+                    // definitely out of range
+
+                    value = result;
+                    return false;
+                }
+
+                if ((source.Length > sizeof(byte)) && (source[..^sizeof(byte)].IndexOfAnyExcept((byte)0x00) != -1))
```

Comment:
Maybe we should try to squeeze in some vectorization of IndexOfAnyExcept for .NET 7?
https://github.com/dotnet/runtime/issues/67942

Reply:
Its not going to be "critical" for this API in .NET 7. I expect these ones will be more rarely used.

If we can manage it, then that will be a bonus :)